File: backend/app/schemas/favorite.py
from datetime import datetime

from pydantic import BaseModel


class FavoriteResponse(BaseModel):
    """收藏记录响应"""

    id: int
    user_id: int
    guide_id: int
    guide_title: str = ""
    guide_author: str = ""
    guide_category: str = ""
    guide_cover: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class FavoriteListData(BaseModel):
    items: list[FavoriteResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/favorite.ts
export interface Favorite {
  id: number;
  user_id: number;
  guide_id: number;
  guide_title: string;
  guide_author: string;
  guide_author_avatar: string | null;
  guide_category: string;
  guide_cover: string | null;
  guide_like_count: number;
  guide_fav_count: number;
  guide_is_liked: boolean;
  guide_is_favorited: boolean;
  created_at: string;
}
